solr/modules/monitor/src/java/org/apache/solr/monitor/search/ReverseSearchComponent.java

```diff
@@ -0,0 +1,260 @@
+    }
+  }
+
+  @Override
+  public String getDescription() {
+    return "Component that integrates with lucene monitor for reverse search.";
+  }
+
+  @Override
+  public void inform(SolrCore core) {
+    core.addCloseHook(
+        new CloseHook() {
+          @Override
+          public void preClose(SolrCore core) {
+            ExecutorUtil.shutdownAndAwaitTermination(executor);
+          }
+        });
+    queryDecomposer = new QueryDecomposer();
+    presearcher = PresearcherFactory.build(core.getResourceLoader(), presearcherParameters);
+    var schema = core.getLatestSchema();
+    if (presearcher instanceof AliasingPresearcher) {
+      String prefix = ((AliasingPresearcher) presearcher).getPrefix() + "*";
+      String maxLengthPattern =
+          Arrays.stream(schema.getDynamicFieldPrototypes())
+              .map(SchemaField::getName)
+              .max(Comparator.comparingInt(String::length))
+              .orElse("");
+      if (!maxLengthPattern.equals(prefix) && prefix.length() <= maxLengthPattern.length()) {
+        throw new IllegalStateException(
+            "Dynamic field pattern "
+                + maxLengthPattern
+                + " is incompatible with presearcher alias pattern "
+                + prefix
+                + ". Increase the length of presearcher alias prefix to be at least"
+                + (maxLengthPattern.length() + 1)
+                + " characters, i.e. set aliasPrefix to "
+                + "_".repeat(maxLengthPattern.length() - DEFAULT_ALIAS_PREFIX.length() + 1)
+                + DEFAULT_ALIAS_PREFIX
+                + " and update the schema accordingly. Refer to documentation on overriding aliasPrefix in "
+                + this.getClass().getSimpleName());
+      }
+      var fieldType = schema.getDynamicFieldType(prefix);
+      if (!fieldType.isTokenized() || !fieldType.isMultiValued()) {
+        throw new IllegalStateException(
+            "presearcher-aliased field must be tokenized and multi-valued.");
+      }
+      if (!fieldType.isTokenized() || !fieldType.isMultiValued()) {
+        throw new IllegalStateException(
+            "presearcher-aliased field must be tokenized and multi-valued.");
+      }
+    }
+
+    for (String fieldName : MonitorFields.REQUIRED_MONITOR_FIELDS) {
+      var field = schema.getFieldOrNull(fieldName);
+      if (field == null) {
+        throw new IllegalStateException(
+            fieldName + " must be defined in schema for saved search to work.");
+      }
+    }
+
+    if (presearcher instanceof TermFilteredPresearcher
+        || presearcher instanceof AliasingPresearcher) {
+      var anyTokenField = schema.getFieldOrNull(MonitorFields.ANYTOKEN_FIELD);
+      if (anyTokenField == null || !anyTokenField.tokenized() || !anyTokenField.multiValued()) {
+        throw new IllegalStateException(
+            MonitorFields.ANYTOKEN_FIELD + " field must be tokenized and multi-valued.");
+      }
+    }
+  }
```

Comment:
cross-reference https://github.com/apache/lucene/pull/13379 w.r.t. making `TermFilteredPresearcher.ANYTOKEN_FIELD` public

Reply:
added https://github.com/apache/solr/pull/2382/commits/28bf2641840f0dfe8f1076661e3387cb03135f7e to use `TermFilteredPresearcher.ANYTOKEN_FIELD`